src/Management/src/EndpointCore/ActuatorRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,131 @@
+                Type _ when endpointType.IsAssignableFrom(typeof(ThreadDumpEndpoint)) => (typeof(ThreadDumpEndpointMiddleware), typeof(IThreadDumpOptions)),
+                Type _ when endpointType.IsAssignableFrom(typeof(ThreadDumpEndpoint_v2)) => (typeof(ThreadDumpEndpointMiddleware_v2), typeof(IThreadDumpOptions)),
+                Type _ when endpointType.IsAssignableFrom(typeof(TraceEndpoint)) => (typeof(TraceEndpointMiddleware), typeof(ITraceOptions)),
+                Type _ when endpointType.IsAssignableFrom(typeof(HttpTraceEndpoint)) => (typeof(HttpTraceEndpointMiddleware), typeof(ITraceOptions)),
+                Type _ when endpointType.IsAssignableFrom(typeof(CloudFoundryEndpoint)) => (typeof(CloudFoundryEndpointMiddleware), typeof(ICloudFoundryOptions)),
+                _ => throw new InvalidOperationException($"Could not find middleware for Type: {endpointType.Name} "),
+            };
+        }
+
+        /// <summary>
+        /// Generic routebuilder extension for Actuators.
+        /// </summary>
+        /// <param name="endpoints">IEndpointRouteBuilder to Map route.</param>
+        /// <typeparam name="TEndpoint">Middleware for which the route is mapped.</typeparam>
+        /// <exception cref="InvalidOperationException">When T is not found in service container</exception>
+        public static void Map<TEndpoint>(this IEndpointRouteBuilder endpoints)
+        where TEndpoint : IEndpoint
+        {
+            if (endpoints == null)
+            {
+                throw new ArgumentNullException(nameof(endpoints));
+            }
+
+            var (middleware, optionsType) = LookupMiddleware(typeof(TEndpoint));
+            var options = endpoints.ServiceProvider.GetService(optionsType) as IEndpointOptions;
+            var mgmtOptionsCollection = endpoints.ServiceProvider.GetServices<IManagementOptions>();
+
+            foreach (var mgmtOptions in mgmtOptionsCollection)
+            {
+                if ((mgmtOptions is CloudFoundryManagementOptions && options is IActuatorHypermediaOptions)
+                    || (mgmtOptions is ActuatorManagementOptions && options is ICloudFoundryOptions))
+                {
+                    continue;
+                }
+
+                var fullPath = options.GetContextPath(mgmtOptions);
+
+                var pipeline = endpoints.CreateApplicationBuilder()
+                    .UseMiddleware(middleware, mgmtOptions)
+                    .Build();
+                var allowedVerbs = options.AllowedVerbs ?? new List<string> { "Get" };
+
+                endpoints.MapMethods(fullPath, allowedVerbs, pipeline);
+            }
+        }
+
+        public static void MapAllActuators(this IEndpointRouteBuilder endpoints, MediaTypeVersion version = MediaTypeVersion.V2)
+        {
+            endpoints.Map<ActuatorEndpoint>();
+            if (Environment.OSVersion.Platform == PlatformID.Win32NT)
```

Comment:
```suggestion
            if (Platform.IsWindows)
```

Reply:
refactored to use Platform.IsWindows everywhere